GVFS/GVFS.FunctionalTests/Should/FileSystemShouldExtensions.cs

```diff
@@ -251,12 +252,54 @@ public DirectoryInfo WithAttribute(FileAttributes attribute)
                 return info;
             }
 
+            private static bool IsMatchedPath(FileSystemInfo info, string repoRoot, string[] prefixes)
+            {
+                if (prefixes == null || prefixes.Length == 0)
+                {
+                    return true;
+                }
+
+                string localPath = info.FullName.Substring(repoRoot.Length + 1);
+
+                if (localPath.Equals(".git", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Include _just_ the .git folder.
+                    // All sub-items are not included in the enumerator.
+                    return true;
+                }
+
+                if (!localPath.Contains(System.IO.Path.DirectorySeparatorChar) &&
+                    (info.Attributes & FileAttributes.Directory) != FileAttributes.Directory)
+                {
+                    // If it is a file in the root folder, then include it.
+                    return true;
+                }
+
+                foreach (string prefixDir in prefixes)
+                {
+                    if (localPath.StartsWith(prefixDir, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return true;
+                    }
+
+                    if (prefixDir.StartsWith(localPath, StringComparison.OrdinalIgnoreCase) &&
+                        Directory.Exists(info.FullName))
+                    {
+                        // For example: localPath = "GVFS" and prefix is "GVFS\\GVFS".

```

Comment:
> prefix is "GVFS\\GVFS"

Do we have an prefixes in our test that are more than one folder deep?

I might have missed it, but as best I could tell everything in `IncludeModeFolders` is a top level folder.

Reply:
Sorry this was my fault.  I didn't pull it [this commit](https://github.com/derrickstolee/VFSForGit/commit/c10e8ba2bf0ba53c6add2749f83f1928169c79b9)